python/tskit/trees.py

```diff
@@ -4842,43 +4837,30 @@ def alignments(
         single byte characters, (i.e., variants must be single nucleotide
         polymorphisms, or SNPs).
 
-        If ``isolated_as_missing`` is True (the default), isolated samples without
-        mutations directly above them will be treated as
-        :ref:`missing data<sec_data_model_missing_data>` and will be
-        represented in the string by the ``missing_data_character``. If
-        instead it is set to False, missing data will be assigned the ancestral state
-        (unless they have mutations directly above them, in which case they will take
-        the most recent derived mutational state for that node).
+        .. note:: :ref:`Missing data<sec_data_model_missing_data>` is not
+            currently supported by this method and it will raise a ValueError
+            if called on tree sequences containing isolated samples.
+            See https://github.com/tskit-dev/tskit/issues/1896 for more
+            information.
 
         See also the :meth:`.variants` iterator for site-centric access
         to sample genotypes and :meth:`.haplotypes` for access to sample sequences
         at just the sites in the tree sequence.
 
         :return: An iterator over the alignment strings for the samples in
             this tree sequence.
-        :param bool isolated_as_missing: If True, the allele assigned to
-            missing samples (i.e., isolated samples without mutations) is
-            the ``missing_data_character``. If False,
-            missing samples will be assigned the ancestral state.
-            Default: True.
-        :param str missing_data_character: A single ascii character that will
-            be used to represent missing data.
-            If any normal allele contains this character, an error is raised.
-            Default: '-'.
+        :param str reference_sequence: The reference sequence to fill in
+            gaps between sites in the alignments.
         :rtype: collections.abc.Iterable
         :raises: ValueError
-            if all coordinates in this tree sequence are not discrete, or
-            if the ``missing_data_character`` exists in one of the alleles,
+            if all coordinates in this tree sequence are not discrete,
             or if the ``reference_sequence`` is not of the correct length.
-        :raises: TypeError if the ``missing_data_character`` or any of the alleles
-            at a site are not a single ascii character.
+        :raises: TypeError if any of the alleles at a site are not a
+            single ascii character.
         """
         if not self.discrete_genome:
             raise ValueError("sequence alignments only defined for discrete genomes")
```

Comment:
NB: in the future, it would be reasonable, I think, to output alignment for non-discrete genomes by filling the spaces between each sit with a "gap" character.

Reply:
I disagree - see point about missing data and discrete coordinates above. There's no point in complicating this any further by bringing in continuous coordinates, it's already going to be very complicated once we've dealt with indels etc. Continuous coordinates are a pretty niche requirement in reality anyway, people who want them are unlikely to also want alignments (and if they do, well, they can figure out what the semantics should be).